### PR TITLE
Fix mosaic preprocessing so it workss for one channel input

### PR DESCRIPTION
**Pull request by Alejandro-Valdes**
# What does this PR do?

modified _update_image so it doesn't remove channel inputs dimension with a squeeze for inputs with only one channel

Fixes [#1464](https://github.com/keras-team/keras-cv/issues/1464)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/keras-team/keras-cv/blob/master/.github/CONTRIBUTING.md),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue? Please add a link
      to it if that's the case.
- [ ] Did you write any new necessary tests?
- [ ] If this adds a new model, can you run a few training steps on TPU in Colab to ensure that no XLA incompatible OP are used?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your  #PR.

### CODE REVIEW

File: keras_cv/layers/preprocessing/mosaic.py [hunk 1]
```diff
@@ -210,8 +210,10 @@ def _update_image(self, images, permutation_order, mosaic_centers, index):
             [input_height, input_width],
         )
         # tf.image.crop_and_resize will always output float32, so we need to recast
-        output = tf.cast(output, self.compute_dtype)
-        return tf.squeeze(output)
+        # tf.image.crop_and_resize outputs [num_boxes, crop_height, crop_width, depth]
+        # since num_boxes is always one we only need the first element
+        output = tf.cast(output[0], self.compute_dtype)
```
Comment (ianstenbit) [inline keras_cv/layers/preprocessing/mosaic.py:215]: Probably a safer approach than indexing at 0 would be to use 
`tf.squeeze(output, axis=0)`
(This will raise an exception if the 0 axis has dimension > 1)

**Review by ianstenbit — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: keras_cv/layers/preprocessing/mosaic.py (hunk 1 above)
Comment (Alejandro-Valdes) [reply keras_cv/layers/preprocessing/mosaic.py]: changed it

**Review by ianstenbit — APPROVED** [review]
Thank you!

**Comment by Alejandro-Valdes** [comment]
thanks @ianstenbit!
